### PR TITLE
fix(touchspin): renamed touchspin to number input, use number type

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/3706

@jessiehuff do you think we should we add `min`, `max`, or anything like that?

### CODE REVIEW

File: src/patternfly/components/NumberInput/number-input-unit.hbs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="pf-c-number-input__unit{{#if number-input-unit--modifier}} {{number-input-unit--modifier}}{{/if}}"
```
Comment (mattnolting) [inline src/patternfly/components/NumberInput/number-input-unit.hbs:1]:
```suggestion
<span class="pf-c-number-input__unit{{#if number-input-unit--modifier}} {{number-input-unit--modifier}}{{/if}}"
```

File: src/patternfly/components/NumberInput/number-input-unit.hbs [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  {{#if number-input-unit--attribute}}
+    {{{number-input-unit--attribute}}}
+  {{/if}}>
+  {{> @partial-block}}
+</div>
```
Comment (mattnolting) [inline src/patternfly/components/NumberInput/number-input-unit.hbs:6]:
```suggestion
</span>
```

**Review by mattnolting — APPROVED** [review]
LGTM! Looks like the `-unit` element needs updated 👍

**Comment by jessiehuff** [comment]
Since `<input type="number">` corresponds to the default role `spinbutton`, aria attributes supported by the spinbutton role should work. However, there are native html attributes that can be used with an `<input type="number">`, and I would default to using those where they exist. [These are the native html attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#Additional_attributes) that are supported with `<input type="number">` which does include `min` and `max`. While I don't think these are required, we could add them as an enhancement or to show their usage for consumers.

**Comment by mcoker** [comment]
@jessiehuff updated the lower/upper threshold examples with `min` and `max`, and added an a11y table with those attrs documented.

**Review by mattnolting — APPROVED** [review]
LPTM

**Review by jessiehuff — APPROVED** [review]
LGTM! 🙂

**Pull request merged**